caproto/curio/server.py

```diff
@@ -29,6 +29,18 @@ def get(self):
         return await super().get()
 
 
+class QueueFull(Exception):
+    ...
+
+
+# Curio queues just block if they are full. We want one that raises.
+class QueueWithFullError(curio.Queue):
+    async def put(self, item):
+        if self.full():
```

Comment:
race condition, but not clear there is a way around that here....

Reply:
I think it's not because there are no threads and no awaits.

Reply:
fair point!